### DISCUSSION TITLE
src/bacnet/basic/object/av.c & ai.c   Alarm_Ack functions  have strange points

### DISCUSSION THREAD

**Discussion by maimomohali** [bacnet-stack/bacnet-stack#636] [General]
In case ai.c, Analog_Input_Alarm_Ack func
 switch (alarmack_data->eventStateAcked) {
    case EVENT_STATE_FAULT:
        if (CurrentAI->Acked_Transitions[TRANSITION_TO_FAULT].bIsAcked   // check fault point of act_transition property
        ~~~
        CurrentAI->Acked_Transitions[TRANSITION_TO_FAULT].bIsAcked = true; // clear the ACK flag

In case av.c, Analog_Value_Alarm_Ack func
 switch (alarmack_data->eventStateAcked) {
    case EVENT_STATE_FAULT:
        if (CurrentAI->Acked_Transitions[**TRANSITION_TO_NORMAL**].bIsAcked   // **why check  "Transition to normal" ???**
        ~~~
        CurrentAI->Acked_Transitions[TRANSITION_TO_FAULT].bIsAcked = true; // clear the ACK flag
    ~~~   
    case EVENT_STATE_NORMAL:
        if (CurrentAV->Acked_Transitions[**TRANSITION_TO_FAULT**].bIsAcked 
        ~~~~ 
                /* Clean transitions flag. */
                CurrentAV->Acked_Transitions[**TRANSITION_TO_NORMAL**].bIsAcked = true //  Checking is  "Fault" & Clearing is "NORMAL"

I think   alarm ack func of ai.c is normal ,but  alarm ack func of av.c is strange.     Is this code  right really?

**Comment by skarg** [comment]
Fixed in #638
